Sprint 11 ver. 1

DmitryBrosalin opened the pull request:
- Добавил возможность добавлять пользователей в друзья, просматривать список друзей, удалять из друзей, а также просматривать общих друзей.
- Добавил возможность ставить лайки на фильмы, удалять их, а также составлять рейтинг по популярности.

DmitryBrosalin committed:
Sprint 11 ver. 1
- Добавил возможность добавлять пользователей в друзья, просматривать список друзей, удалять из друзей, а также просматривать общих друзей.
- Добавил возможность ставить лайки на фильмы, удалять их, а также составлять рейтинг по популярности.

DmitryBrosalin committed:
Sprint 11 ver. 1
- Добавил возможность добавлять пользователей в друзья, просматривать список друзей, удалять из друзей, а также просматривать общих друзей.
- Добавил возможность ставить лайки на фильмы, удалять их, а также составлять рейтинг по популярности.

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,51 +1,59 @@
+    @PutMapping(value = "/{id}/like/{userId}")
+    public void addLike(@PathVariable String id,
+                        @PathVariable String userId) {
+        try {
```
`try/catch` лучше вынести внутрь сервиса
Замечание для всех методов

slavrikk replied:
Если сделать тип Long, то этого парсинга не будет

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -1,59 +1,83 @@
+
+    @GetMapping(value = "/{id}")
+    public User findUser(@PathVariable String id) {
+        try {
```
Аналогичный комментарий

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,51 +1,59 @@
+    }
+
+    @PutMapping(value = "/{id}/like/{userId}")
+    public void addLike(@PathVariable String id,
```
Можно сразу id обозначить как Long `@PathVariable Long id`

DmitryBrosalin replied:
Понял, спасибо! Просто в теории на курсе не было прописано это - всегда через Парсинг делали. Я подумал, что это "золотой стандарт". Но, видимо, нет) учту, исправлю 👌🏻

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -1,59 +1,83 @@
+    }
+
+    @GetMapping(value = "/{id}")
+    public User findUser(@PathVariable String id) {
```
Аналогичный комментарий

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -18,4 +19,10 @@ public class Film {
     LocalDate releaseDate;
     @Positive
     int duration;
+    Set<Long> likes;
```
Лучше сразу инициализировать коллекцию

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -18,4 +19,5 @@ public class User {
     private String name;
     @PastOrPresent
     private LocalDate birthday;
+    private Set<Long> friends;
```
Лучше сразу инициализировать коллекцию

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/InMemoryFilmStorage.java:
```diff
@@ -0,0 +1,70 @@
+
+    @Override
+    public Collection<Film> getPopularFilms(long size) {
+        return films.values().stream()
```
Не учитывается результирующий размер коллекции `size`

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/InMemoryFilmStorage.java:
```diff
@@ -0,0 +1,70 @@
+    @Override
+    public Film createFilm(Film film) {
+        film.setId(getNextId());
+        film.setLikes(new HashSet<>());
```
Если инициализировать коллекцию, то этих вставок не будет

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/InMemoryFilmStorage.java:
```diff
@@ -0,0 +1,70 @@
+            throw new NotFoundException("Фильм с id = " + film.getId() + " не найден.");
+        }
+        if (film.getLikes() == null) {
+            film.setLikes(new HashSet<>());
```
Если инициализировать коллекцию, то этих вставок не будет

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/storage/user/InMemoryUserStorage.java:
```diff
@@ -0,0 +1,63 @@
+            user.setName(user.getLogin());
+        }
+        user.setId(getNextId());
+        user.setFriends(new HashSet<>());
```
Если инициализировать коллекцию, то этих вставок не будет

DmitryBrosalin committed:
Sprint 11 ver. 2
- Поменял тип переменной пути id на long, убрал парсинг.
- Инициализация коллекций likes и friends происходит сразу.
- При выводе популярных фильмов теперь учитывается размер.
